Use types from std::os::raw instead of libc where possible

Boscop opened the pull request:
Now plugins don't need to depend on libc for `Plugin::open()`.

I also made `load_pointer` uppercase (`LOAD_POINTER`) because it's a global var and caused a warning.

Boscop committed:
use types from std::os::raw instead of libc where possible. now plugins don't need to depend on libc

overdrivenpotato commented:
Tests appear to be failing, I'll merge it in if you can fix the build

Boscop committed:
fix

overdrivenpotato commented on src/interfaces.rs:
```diff
@@ -70,9 +71,9 @@ pub fn dispatch(effect: *mut AEffect, opcode: i32, index: i32, value: isize, ptr
+    let copy_string = |string: &String, max: usize| {
         unsafe {
-            libc::strncpy(ptr as *mut c_char,
+            strncpy(ptr as *mut c_char,
```
Let's leave this line as `libc::strncpy` to make it clear that we are not implementing our own.

overdrivenpotato reviewed (changes requested):
Missed this earlier, looks good otherwise

Boscop committed:
removed allocations in copy_string

Boscop commented:
Both local closures of `copy_string` were the same (from copy pasting last year) so I moved it out into a function and removed the allocations: `string.clone()` and `CString::new()` were both allocating and walking the string.

Boscop committed:
sine_synth now reports correctly that it can receive midi events. also changed crate type to cdylib

monomadic commented:
Can we merge this asap as I'm waiting on this fix.

Perhaps boscop should have the ability to merge PRs?

zyvitski commented:
I second the motion to merge.

overdrivenpotato committed:
Add proper docs to function

overdrivenpotato commented:
I'll try and be quicker regarding future PRs, the last few days were fairly busy for me.
